services/supervisor/src/api.rs

```diff
@@ -79,7 +79,13 @@ impl PrivateApi for ApiImpl<'_> {
     }
 
     fn propose_config(&self, proposal: ConfigPropose) -> Result<Hash, Self::Error> {
-        self.broadcast_transaction(proposal).map_err(From::from)
+        // Discard proposes whose `actual from` heights are same with already registered proposes
+        if Schema::new(self.0.instance.name,self.0.snapshot())
+            .pending_propose_hashes().contains(&proposal.actual_from.0) {
+            Err(Self::Error::from(failure::format_err!("Config proposal with the same height already registered")))
```

Comment:
```suggestion
            Err(Self::Error::from(failure::format_err!("Config proposal with the same height has already been registered")))
```

Reply:
Maybe split this whole block into several lines? It's hard to read (and indentation looks a bit weird).